backend/app/api/routes/fine_tuning.py

```
@@ -179,22 +185,72 @@ def process_fine_tuning_job(
+
+    # Validate system prompt
+    if not system_prompt.strip():
+        raise HTTPException(
+            status_code=400, detail="System prompt must be a non-empty string"
+        )
+
+    # Validate file is CSV
+    if not file.filename.lower().endswith(".csv"):
+        raise HTTPException(status_code=400, detail="File must be a CSV file")
+
     client = get_openai_client(  # Used here only to validate the user's OpenAI key;
```

Ruff (0.13.1)

Local variable `client` is assigned to but never used

Remove assignment to unused variable `client`

(F841)